src/functionalTest/java/uk/gov/hmcts/reform/sendletter/controllers/SendLetterTest.java

```diff
@@ -42,15 +43,15 @@
     @Autowired
     private LetterChecksumGenerator generator;
 
-    @SpyBean
+    @MockBean
```

Comment:
that's functional test, letter service should not be mocked. mock only directly related third party clients/services, like azureQueueClient below